command/quota_apply.go

```diff
@@ -280,9 +281,46 @@ func parseQuotaResource(result *api.Resources, list *ast.ObjectList) error {
 		return err
 	}
 
+	// Manually parse
+	delete(m, "devices")
+
 	if err := mapstructure.WeakDecode(m, result); err != nil {
 		return err
 	}
 
+	// Parse devices
+	if o := listVal.Filter("device"); len(o.Items) > 0 {
+		result.Devices = make([]*api.RequestedDevice, 0)
+		if err := parseDeviceResource(&result.Devices, o); err != nil {
+			return multierror.Prefix(err, "devices ->")
+		}
+	}
+
+	return nil
+}
+
+func parseDeviceResource(result *[]*api.RequestedDevice, list *ast.ObjectList) error {
+	for _, o := range list.Elem().Items {
+		// Check for invalid keys
+		valid := []string{
+			"name",
+			"count",
```

Comment:
can also have `constraint` and `affinity`, no?

https://developer.hashicorp.com/nomad/docs/job-specification/device#device-parameters

Reply:
it can, but do we want to include those in the quota spec? I understand what `constraint` would enable operators to do, but how would `affinity` work with a device quota?

Reply:
or are you suggesting to just accept them as valid input but simply ignore at quota enforcement?

Reply:
Oh sorry I was thinking this was a normal resources block, but it's a quota block ... where these fields wouldn't exist